Index: src/main/java/io/github/benas/sbi/JobConfiguration.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/io/github/benas/sbi/JobConfiguration.java	(revision 74d443e914760a64c8559bc04e45ebcf0ced511a)
+++ src/main/java/io/github/benas/sbi/JobConfiguration.java	(revision 74d443e914760a64c8559bc04e45ebcf0ced511a)
@@ -0,0 +1,44 @@
+package io.github.benas.sbi;
+
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.batch.core.launch.support.SimpleJobLauncher;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.core.repository.support.JobRepositoryFactoryBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import javax.sql.DataSource;
+
+@Configuration
+@Import(DataSourceConfiguration.class)
+public class JobConfiguration {
+
+	private DataSource dataSource;
+
+	private PlatformTransactionManager transactionManager;
+
+	@Autowired
+	public JobConfiguration(DataSource dataSource, PlatformTransactionManager transactionManager) {
+		this.dataSource = dataSource;
+		this.transactionManager = transactionManager;
+	}
+
+	@Bean
+	public JobRepository jobRepository() throws Exception {
+		JobRepositoryFactoryBean jobRepositoryFactoryBean = new JobRepositoryFactoryBean();
+		jobRepositoryFactoryBean.setDataSource(dataSource);
+		jobRepositoryFactoryBean.setTransactionManager(transactionManager);
+		return jobRepositoryFactoryBean.getObject();
+	}
+
+	@Bean
+	public JobLauncher jobLauncher() throws Exception {
+		SimpleJobLauncher simpleJobLauncher = new SimpleJobLauncher();
+		simpleJobLauncher.setJobRepository(jobRepository());
+		return simpleJobLauncher;
+	}
+
+}
